### PR TITLE
fix: Fix bug with 'argo delete --older'

### LINKED ISSUE #3695: Argo CLI - argo delete --older no longer works
## Summary 

--older flag in `argo delete` no longer works. It used to work with previous versions of argo CLI/ argo controller

## Diagnostics

Argo v2.9.2
Argo CLI - v2.9.2

Please refer below:

image

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by simster7**
Fixes #3695

### CODE REVIEW

File: cmd/argo/commands/delete.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func NewDeleteCommand() *cobra.Command {
 					ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: flags.namespace},
 				})
 			}
-			if all || flags.completed || flags.resubmitted || flags.prefix != "" || flags.labels != "" {
+			if all || flags.completed || flags.resubmitted || flags.prefix != "" || flags.labels != "" || flags.finishedAfter != "" {
```
Comment (alexec) [inline cmd/argo/commands/delete.go:48]: test?

**Comment by alexec** [comment]
I'd like to understand why `TestWorkflowDeleteOlder`.

**Comment by simster7** [comment]
> I'd like to understand why TestWorkflowDeleteOlder did not catch this.

https://github.com/argoproj/argo/issues/3695#issuecomment-671080226

File: cmd/argo/commands/delete.go (hunk 1 above)
Comment (simster7) [reply cmd/argo/commands/delete.go:48]: Test already exists

**Comment by alexec** [comment]
Available in [v2.10.0-rc7](https://github.com/argoproj/argo/releases/tag/v2.10.0-rc7).

**Pull request merged**